test/detail/test_function.cpp

```diff
@@ -158,6 +158,56 @@ BOOST_FIXTURE_TEST_CASE(triggering_several_once_expectations_is_valid, mock_erro
     }
 }
 
+BOOST_FIXTURE_TEST_CASE(string_likes_are_matched, mock_error_fixture)
+{
+    const char* c_string = "value";
+    const char c_string2[] = "value";
+    BOOST_REQUIRE(c_string != c_string2); // Different pointers
```

Comment:
nit: I think the comment is a little redundant, given the code clearly says the same thing 😄

Reply:
True. Improved on that after partially reverting it due to exactly this check failing ;-)